source/extensions/filters/http/api_key_auth/api_key_auth.h

```diff
@@ -0,0 +1,137 @@
+#pragma once
+
+#include "envoy/extensions/filters/http/api_key_auth/v3/api_key_auth.pb.h"
+#include "envoy/extensions/filters/http/api_key_auth/v3/api_key_auth.pb.validate.h"
+#include "envoy/stats/stats_macros.h"
+
+#include "source/common/common/logger.h"
+#include "source/extensions/filters/http/common/pass_through_filter.h"
+
+#include "absl/container/flat_hash_map.h"
+#include "absl/container/flat_hash_set.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace ApiKeyAuth {
+
+/**
+ * All Basic Auth filter stats. @see stats_macros.h
+ */
+#define ALL_API_KEY_AUTH_STATS(COUNTER)                                                            \
+  COUNTER(allowed)                                                                                 \
+  COUNTER(unauthorized)                                                                            \
+  COUNTER(forbidden)
+
+/**
+ * Struct definition for API key auth stats. @see stats_macros.h
+ */
+struct ApiKeyAuthStats {
+  ALL_API_KEY_AUTH_STATS(GENERATE_COUNTER_STRUCT)
+};
+
+using ApiKeyAuthProto = envoy::extensions::filters::http::api_key_auth::v3::ApiKeyAuth;
+using ApiKeyAuthPerScopeProto =
+    envoy::extensions::filters::http::api_key_auth::v3::ApiKeyAuthPerScope;
+
+using ApiKeyMap = absl::flat_hash_map<std::string, std::string>;
+
+class KeyResult {
```

Comment:
Seems that a struct needs to be used here.

Please add comments for all the classes and non-private methods.

Reply:
SGTM.